pytest_fixtures/sys_fixtures.py

```diff
@@ -32,3 +33,39 @@ def proxy_port_range(default_sat):
         port_pool_range = settings.fake_capsules.port_range
         if default_sat.execute(f'semanage port -l | grep {port_pool_range}').status != 0:
             default_sat.execute(f'semanage port -a -t websm_port_t -p tcp {port_pool_range}')
+
+
+@pytest.fixture
+def register_to_dogfood():
+    pass
+
+
+# TODO install from dogfood, path should be saved in conf/repos and in settings.repos.dogfood
+# settings.repos.dogfood_repo_host
+# yum -y localinstall <dogfood url>
+# TODO register satellite by command register_contenthost
+#  with org and activation key from command below
+# TODO parametrize it for each satellite version
+# subscription-manager register --org <dogfood org> --activationkey <dogfood act-key>
+# TODO after registration install cockpit plugin can be started
+
+# TODO BZ needs to be mentioned otherwise I would forget
+# instead of  'foreman-maintain packages install -y tfm-rubygem-foreman_remote_execution-cockpit'
+# 'foreman-installer --enable-foreman-plugin-remote-execution-cockpit' should be used
+
+
+@pytest.fixture
+def install_cockpit_plugin(default_sat, register_to_dogfood):
+    cmd_result = default_sat.execute(
+        'foreman-maintain packages install -y tfm-rubygem-foreman_remote_execution-cockpit'
+    )
+    if cmd_result.status != 0:
+        raise SatelliteHostError(
+            f'Error during cockpit installation, installation output: {cmd_result.stdout}'
+        )
+    cmd_result = default_sat.execute(
+        f'sshpass -p "{settings.server.ssh_password}" ssh-copy-id \
```

Comment:
I know it is not a concern now, but note that passing password on command-line is not considered safe as any user/process can read it (`ps aux`).

Reply:
Thanks for your comment. Sure you are right. 
But I think we do not care. Because no evil doer should be able to log into our machine.
We already use the same principle in` tests/robottelo/test_ssh.py `
`tests/foreman/uitest_ldap_authentication.py`
`robottelo.ssh.py`
`virtwtho_utils.py`

Reply:
I got you both @jaryn and @tstrych. However, I simply do not like the implementation that is using plaintext passwords. Looking at the `sshpass` man page, there is a `-f` option that allows us to pass the path to a file containing the password instead of passing password directly.
Every SatLab VM has `id_rsa_hudson` key installed in the `authorized_keys`, that means we can make use of it and use pubkey auth instead of the current solution.
Is that doable for all the modules that you mentioned @tstrych?